src/support/string.cpp

```diff
@@ -482,10 +484,7 @@ std::vector<char> unescapeJSONToWTF8(const char* str) {
     // Write out the results.
     unescaped.push_back(x & 0xff);
     x >>= 8;
-    if (x) {
-      unescaped.push_back(x);
-    }
-    // TODO UTF stuff
+    unescaped.push_back(x);
```

Comment:
This does not correctly encode code points that require surrogate pairs. Those code points require 4 bytes to encode.

Reply:
Test added in https://github.com/WebAssembly/binaryen/pull/7415

Reply:
Thanks for the test!

See the last commit, I added that test here for string lifting. It looks correct to me?

Reply:
Oh, right, because `\uXXXX` cannot express a code point in the range that needs surrogates to begin with. So I think this code is correct, but I still think it would be nicer if it used the existing `writeWTF16CodePoint` helper. WDYT?

Reply:
Sure, done.